src/test/java/roomescape/repository/reservationtime/ReservationTimeH2RepositoryTest.java

```diff
@@ -0,0 +1,67 @@
+package roomescape.repository.reservationtime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.LocalTime;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.JdbcTest;
+import org.springframework.context.annotation.Import;
+import roomescape.domain.ReservationTime;
+
+@JdbcTest
+@Import(ReservationTimeH2Repository.class)
+class ReservationTimeH2RepositoryTest {
+
+    @Autowired
+    private ReservationTimeH2Repository reservationTimeH2Repository;
+
+    private ReservationTime reservationTime;
+
+    @BeforeEach
+    void init() {
+        reservationTime = reservationTimeH2Repository.save(
+                new ReservationTime(null, LocalTime.of(12, 0))
+        );
+    }
+
+//    @Test
+//    @DisplayName("ReservationTime을 저장한다.")
+//    void save() {
+//        ReservationTime reservationTime = new ReservationTime(null, LocalTime.of(12, 0));
+//
+//        ReservationTime save = reservationTimeH2Repository.save(reservationTime);
+//
+//        assertThat(save).isEqualTo(ReservationTime.of(1L, reservationTime));
+//    }
+
+    @Test
+    @DisplayName("id에 맞는 ReservationTime을 제거한다.")
+    void delete() {
+        reservationTimeH2Repository.delete(reservationTime.id());
+
+        assertThat(reservationTimeH2Repository.findAll()).hasSize(0);
+    }
+
+    @Test
+    @DisplayName("모든 ReservationTime을 찾는다.")
+    void findAll() {
+        reservationTimeH2Repository.save(new ReservationTime(null, LocalTime.of(12, 0)));
+        reservationTimeH2Repository.save(new ReservationTime(null, LocalTime.of(11, 0)));
+
+        List<ReservationTime> found = reservationTimeH2Repository.findAll();
+
+        assertThat(found).hasSize(3);
+    }
+
+    @Test
+    @DisplayName("id에 맞는 ReservationTime을 찾는다.")
+    void findBy() {
+        ReservationTime found = reservationTimeH2Repository.findById(reservationTime.id());
```

Comment:
> 데이터를 롤백하는 방법도 있지만 id를 직접 집어넣는 방법도 있어요. 현재는 save 메소드를 호출해서 저장하고 있는데 sql문 자체를 init하는 방법이 있습니다.

위 방법이 신세계네요... 말씀해주신대로 save 메서드에 의존하지 않고 초기값 셋팅이 가능하고, DB와 id값이 매번 초기화돼서 테스트를 진행하기 정말 편리한 것 같습니다.

id를 지정해서 넣어주지 않으면 매번 테스트마다 id값이 1부터 다시 시작해서 혹시 필요한 경우엔 id값으로도 테스트가 가능할 것 같습니다! 

> 어노테이션을 사용하실때에는 어떤 기능이 있는지를 보고 쓰는 것도 좋지만 어떤 이유에서 무엇때문에 트리거가 되어서 동작하는지 파악해두는것이 더 중요하기 때문에 사용하실때 확인해보시면 좋겠네요.

위 말씀에 대해 동감해서 `@JdbcTest` 어노테이션에 대해 간략히 살펴보았습니다! 사실 스프링프레임워크가 어노테이션을 보고 어떻게 해석해서 어떤식으로 동작하는지는 아직 잘 모르겠어요. 그래서 어노테이션 코드를 보고 아래와 같이 동작하지 않을까 추측해봤는데, 오리가 의도한 학습방향이 맞을까요?!
![jdbcTest](https://github.com/woowacourse/spring-roomescape-admin/assets/146502065/c8b6e129-182b-4cc5-9e55-dee64f818fea)

Reply:
왜 롤백이 되느냐는 내용에 대한 이야기이긴했어요. 말씀해주신 내용도 `@JdbcTest`의 구조에 대한 이야기라서 잘 정리해주신 것 같아요.

https://docs.spring.io/spring-boot/docs/1.5.6.RELEASE/reference/html/boot-features-testing.html
> If your test is @Transactional, it will rollback the transaction at the end of each test method by default. If you’re using this arrangement in combination with either RANDOM_PORT or DEFINED_PORT, any transaction initiated on the server won’t rollback as the test is running in a different thread than the server processing.

롤백은 `@JdbcTest` 어노테이션의 `@Transactional`때문인데요. 다음미션 혹은 다다음 미션에서 사용될 개념이니 이런 내용이 있다 정도로 넘어가주시면 좋을 것 같아요.

Reply:
> 왜 롤백이 되느냐는 내용에 대한 이야기이긴했어요.

그렇네요 제가 `쿼리를 롤백해주는 기능이 있다고 배웠습니다.`라고  언급해놓고 다른 얘기를 하고 있었네요 ㅎㅎ.. 남겨주신 링크와 인용문 잘 읽어보았습니다!  덕분에 `@SpringBootTest`를 사용할 때 속성 정보를 어떻게 사용하는지에 따라 `@Transcational`이 의도한대로 transcation을 롤백해줄 수 있는지 여부가 나뉠 수도 있다는 사실까지 학습할 수 있었어요 :) 다음 미션들에서 배울 기회가 되면 좀 더 깊게 파보겠습니다!

> 어노테이션을 사용하실때에는 어떤 기능이 있는지를 보고 쓰는 것도 좋지만 어떤 이유에서 무엇때문에 트리거가 되어서 동작하는지 파악해두는것이 더 중요하기 때문에 사용하실때 확인해보시면 좋겠네요.

제게 큰 도움이 되는 조언인 것 같아요. 앞으로도 꼭 명심하도록 하겠습니다 !!